src/Resources/views/Block/block_search_result.html.twig

```diff
@@ -49,8 +49,12 @@ file that was distributed with this source code.
                             {% set link = admin.getSearchResultLink(result) %}
                             {% if link %}
                                 <li><a href="{{ link }}">{{ admin.toString(result) }}</a></li>
+                                {# NEXT_MAJOR: Remove previous line and uncomment the following one #}
+                                {# <li><a href="{{ link }}">{{ result|subjectAsString(admin)|raw }}</a></li> #}
```

Comment:
Ideally, `subjectAsString` should be able to work with just an admin code (so a string), so that we are not forced to have this responsability in the admin class. The Twig extension should be able to find the `AdminExtractor` implementation for that admin code without asking the admin object.

Reply:
I decided to let the responsibility at the admin class in order to allow the user override the method to control the string representation for an admin's subject (keeping the current behavior).
I'm not against on remove this method from the admin class if we have another way to provide the same feature.

Reply:
I remember facing the same issue when extracting the breadcrumbs builder responsabilité, and I think there was something similar for the template registry system. Sorry for being vague, I'm on my phone. IMO a solution would be to have the same service being used for 90% of admins (one admin being identified by its admin code), and then the 10% that need "overrides" could use other implementations. So 1 admin has 1 admin code, many services that may or may not be shared with other admins, and hopefully, in the end, no God admin object.